Add Grep exercise

tofische opened the pull request:
Added a new Haskell practice exercise Grep based on the available problem specification.

tofische commented:
Can I contribute?

ErikSchierboom commented on exercises/practice/grep/.docs/instructions.append.md:
```diff
@@ -0,0 +1,4 @@
+# Hints
```
```suggestion
# Instructions append
```

ErikSchierboom commented on exercises/practice/grep/.docs/instructions.append.md:
```diff
@@ -0,0 +1,4 @@
+# Hints
+
+To simplify the exercise the flags are already given parsed and input files read,
```
```suggestion
To simplify the exercise the flags are already parsed and input files read,
```

tofische committed:
Add Grep exercise

tofische committed:
Updated config.json to reference proper package.yaml

tofische committed:
Use exemplar instead of example for concept exercises.

keiravillekode committed:
bin/verify-exercises-in-docker script (#1295)

* bin/verify-exercises-in-docker script

* skip deprecated exercises

BNAndras committed:
Fix typo in TESTS.md (#1284)

dependabot[bot] committed:
build(deps): bump actions/checkout from 4.1.7 to 6.0.2 (#1281)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.7 to 6.0.2.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/692973e3d937129bcbf40652eb9f2f61becf3332...de0fac2e4500dabe0009e67214ff5f5447ce83dd)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: 4.2.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

keiravillekode committed:
ci uses ubuntu-24.04 (#1290)

* ci uses ubuntu-24.04

* actions/cache@5a3ec84eff668545956fd18022155c47e93e2684

* yq v4.46.1

---------

Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/cache from 4.3.0 to 5.0.5 (#1280)

Bumps [actions/cache](https://github.com/actions/cache) from 4.3.0 to 5.0.5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/0057852bfaa89a56745cba8c7296529d2fc39830...27d5ce7f107fe9357f9df03efb73ab90386fccae)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.1.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tomas Fischer <tofische@users.noreply.github.com>

meatball133 committed:
Rework Bools concept (#1278)

* Rework bool concept

* Remove Crystal mention

* Remove vsc file

IsaacG committed:
Use `Test.Hspec.Runner` to run concept tests, matching what is done for practice tests (#1297) [no important files changed]

tofische committed:
Merge branch 'main' into grep-exercise

keiravillekode committed:
verify-exercises-in-docker: avoid cleanup failure (#1303)

IsaacG committed:
Bump resolver to `lts-22.44` to match the test runner (#1304)

* Bump resolver to `lts-22.44` to match the test runner

* Update examples to pass testing

[no important files changed]

tofische committed:
Add grep to config.json

tofische committed:
Satisfy hlint

tofische committed:
Update resolver

tofische committed:
Satisfy hlint

tofische committed:
Add type signatures

tofische committed:
Add type signatures

tofische commented:
@MatthijsBlom I've incorporated IO
